CMakeLists.txt

```diff
@@ -30,9 +30,35 @@ endif(NOT DEFINED WANT_EIO)
 aux_source_directory("${CMAKE_CURRENT_SOURCE_DIR}/src" EVFIBERS_SOURCES)
 aux_source_directory("${CMAKE_CURRENT_SOURCE_DIR}/coro" CORO_SOURCES)
 
+# OSX-related checks - linking, using ucontext, etc. is slightly different than
+# on linux
+if(APPLE)
+	# def needed to get ucontext support
+	set(CMAKE_REQUIRED_DEFINITIONS "${CMAKE_REQUIRED_DEFINITIONS} -D_XOPEN_SOURCE")
+	add_definitions(-D_XOPEN_SOURCE)
+	# ucontext is labeled deprecated, so make sure -Werror doesn't kill the
+	# compile
+	set(CMAKE_C_FLAGS "-Wno-error=deprecated-declarations ${CMAKE_C_FLAGS}")
+	# OSX uses MAP_ANON for mmap while linux uses MAP_ANONYMOUS
+	set(FBR_MAP_ANON_FLAG MAP_ANON)
+	set(EIO_LD_WHOLE_ARCHIVE -Wl,-all_load)
+	set(EIO_LD_NOWHOLE_ARCHIVE )
+else()
```

Comment:
Same here, please specify condition to be consistent with the rest of the CMakeLists sources.